unittests/test_check_filters.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+
+import reframe.core.runtime as rt
+import reframe.frontend.check_filters as filters
+import unittests.fixtures as fixtures
+from reframe.core.systems import SystemPartition
+from reframe.core.pipeline import RegressionTest
+from reframe.utility.typecheck import Str
+
+class TestCheckFilters(unittest.TestCase):
+    def setUp(self):
+        #TODO: use correct initialization
+        #check 1
+        self.check1 = RegressionTest()
+        self.check1.name = 'check1'
+        self.check1.tags = {'a', 'b', 'c', 'd'}
+        self.check1.valid_prog_environs = ['env1', 'env2']
+        self.check1.valid_systems = ['testsys:gpu', 'testsys2:mc']
+        self.check1.num_gpus_per_node = 1
+        #check 2
+        self.check2 = RegressionTest()
+        self.check2.name = 'check2'
+        self.check2.tags = {'x', 'y', 'z'}
+        self.check2.valid_prog_environs = ['env3']
+        self.check2.valid_systems = ['testsys:mc', 'testsys2:mc']
+        self.check2.num_gpus_per_node = 0
+
+    def test_have_name(self):
+        self.assertTrue(filters.have_name('check1')(self.check1))
```

Comment:
This is the correct way to test this, because it reflects the way we are using the filter functions.

```python
import reframe.utility.sanity as sn

self.assertEqual(1, sn.count(filter(filters.have_name('check1'), self.checks)))
```

The reason that you cannot use `len()` directly on the result of the `filter()` function is because `filter()` returns an generator, so you would have to create a concrete data structure (e.g., `list`) out of it first. `sn.count` handles this case automatically, but it consumes the generator (not a problem in our case). It also returns a deferrable expression, but since it's included in boolean check, it will be automatically evaluated.